src/main/java/com/hivemq/cli/hivemq/policies/ListPoliciesTask.java

```diff
@@ -39,59 +40,82 @@ public class ListPoliciesTask {
     private final @Nullable String topic;
     private final @Nullable String @Nullable [] policyIds;
     private final @Nullable String @Nullable [] schemaIds;
+    private final @Nullable String @Nullable [] fields;
+    private final @Nullable Integer limit;
 
     public ListPoliciesTask(
             final @NotNull OutputFormatter outputFormatter,
             final @NotNull PoliciesApi policiesApi,
             final @Nullable String topic,
             final @Nullable String @Nullable [] policyIds,
-            final @Nullable String @Nullable [] schemaIds) {
+            final @Nullable String @Nullable [] schemaIds,
+            final @Nullable String @Nullable [] fields,
+            final @Nullable Integer limit) {
         this.outputFormatter = outputFormatter;
         this.policiesApi = policiesApi;
         this.topic = topic;
         this.policyIds = policyIds;
         this.schemaIds = schemaIds;
+        this.fields = fields;
+        this.limit = limit;
     }
 
     public boolean execute() {
-        final String policyIdsParameter;
+        final String fieldsQueryParam;
+        if (fields == null) {
+            fieldsQueryParam = null;
+        } else {
+            fieldsQueryParam = String.join(",", fields);
+        }
+
+        final String policyIdsQueryParam;
         if (policyIds == null) {
-            policyIdsParameter = null;
+            policyIdsQueryParam = null;
         } else {
-            policyIdsParameter = String.join(",", policyIds);
+            policyIdsQueryParam = String.join(",", policyIds);
         }
 
-        final String schemaIdsParameter;
+        final String schemaIdsQueryParam;
         if (schemaIds == null) {
-            schemaIdsParameter = null;
+            schemaIdsQueryParam = null;
         } else {
-            schemaIdsParameter = String.join(",", schemaIds);
+            schemaIdsQueryParam = String.join(",", schemaIds);
         }
 
-        final List<Policy> allPolicies = new ArrayList<>();
+        List<Policy> allPolicies = new ArrayList<>();
 
         try {
             String nextCursor = null;
```

Comment:
Yeah I thought about that but I don't think it's worth it since there's a lot of differences between the uses.

I could do something like this: https://gist.github.com/busybeetree/5715ba536025a9be5b38c1875eafb90b
but it would only be usable with list schemas and policies, not client IDs, since they're collected differently.